### ISSUE TITLE
[API Design](Cache): Add `implicitDependencies` option to `keepFresh`

### ISSUE THREAD

**Issue by NazariiShvets** [igorkamyshev/farfetched#350]
Precondition: [docs](https://farfetched.pages.dev/recipes/cache.html#hashing-algorithm)
```
So, the key is a hash of the following data:
- SID of the Query
- params of the particular call of the Query
- current values of all external Stores that affect Query
To get short and unique key, we stringify all data, concatenate it and then hash it with SHA-1.
```

Usually we have a lots of queries which is tied to user session. Common usecases: 
 - active user organization (in cases where app have two-layer auth).  
 - current role(permissions) of user (backend can in runtime change it)

But there is no api to provide this knowledge for cache (it usually does not passed in params and backend takes it from session tokens)

**Comment by igorkamyshev** [comment]
I like the idea 💙

But how is it connected with `keepFresh`? I assume it is about `cache`?

**Comment by NazariiShvets** [comment]
I suggest it as `keepFresh` parameter bcs its what first come to my mind when I tried to connect my session info to `keepFresh`

Sure, its more about `cache` than about `keepFresh`. So if you think it better suited there I think it ok

**Comment by igorkamyshev** [comment]
`keepFresh` already can use implicit deps. Just put it in `triggers` field 😇 https://farfetched.pages.dev/api/operators/keep_fresh.html#keepfresh-query-config

So, what this new field should do to `cache`? I do not get it.

**Comment by NazariiShvets** [comment]
I think its important to account for this in cache key, bcs it could lead to unnessesary cache hits (bcs when i switching my accounts, and see cached data from previous account it doesnt looks right)

**Comment by igorkamyshev** [comment]
So, why you do not want to use smth like this:
```ts
sample({
  clock: switchAccount,
  target: purgeCache,
})

cache(query, {
  purge: purgeCache
})
```

https://farfetched.pages.dev/api/operators/cache.html#formulae

**Comment by NazariiShvets** [comment]
Sure u can purge it. And its the only(for now) way to deal with this type of cases

My point is — add ability to preserve that cache than your depencenies changing instead of purging that data, and hit it when dependency changed back(if cache exist, so u dont need to load that data again if it still fresh)

**Comment by igorkamyshev** [comment]
So, we talk about something like this?

```ts 
cache(query, {
  extraDependencies: [$userStatus]
})
```

But I still do not understand, how this extra dep can influence the result of the Query if you do not pass it to the Query?

```ts 
const query = createJsonQuery({
  request: {
    headers: combine($userStatus, ...)
  }
})
```

In this case, $userStatus would be used as dependency for cache key generation automatically.

**Comment by NazariiShvets** [comment]
I spend some time digging codebase, and from that i could tell, i easier to add `externalDependencies` to `createQuery` itself, rather than making `query.__.sourced` reactive (and we need to make it reactive, so order of attaching `cache`, `keepFresh`  doesnt matter)

Also it should be easy to propagate this param to `createJsonQuery`

So it seems that this issue should be renamed to `... option to createQuery` 🙃 

What u think about that?

**Comment by igorkamyshev** [comment]
We do not talk about "easier" ☺️ I'm trying to extract information about use cases. Could you answer my ques from the previous comment?

**Comment by igorkamyshev** [comment]
I'll thing about this case.

**Comment by igorkamyshev** [comment]
I believe, you can achieve this behavior with custom factory with passing your extraDepa to sourced of createHeadlessQuery.

Does it work for you?

**Comment by NazariiShvets** [comment]
Yes, this is implementation to which i come trying to bring this feature in

But it too low-level to use as default factory for all queries. For my cases, where i can replace `createQuery` with `createHeadless query` and duplicate code of `createQuery` inside my codebase it can be done. But if I would build my app using high-level abstractions like `createJsonQuery` it would be devastating to fallback all factories to `createHeadlessQuery`

**Comment by igorkamyshev** [comment]
So, just write createNazarQuery in your own code and use it everywhere. I see not reasons to bring it to the core.

**Comment by igorkamyshev** [comment]
I renamed this issue to reflect it ☺️ I'll write the recipe.

**Comment by sandeep661** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1500 in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @chapskev, @MeyCry, @newbalancem5, @ankormoreankor, @PaulineNemchak, @embyth, @LauraNicholas, @Evdiso, @batogov, @baybara-pavel, @devmargooo, @brachkow, @neverlous, @kkoscielniak, @uttamkoirala, @deniskorotin12, @bruna17, @milovidov983, @Songs0fFailure, @Arly0, @3r7k, @john-ea, @pavel-alay, @Daniel-Aristov, @Rubio6321, @AnnaKulib, @toliklunev, @Avdushin, @axosipenko, @christosgiamakis, @nieopierzony, @gobacktosleep, @mrstatic23, @evasilev, @mattnix4, @burmisha, @VictorNov, @VDGK, @DGryanko, @oxmap, @Mohamed988o, @MikVed, @gord-93, @AlexandrPristupa, @nezvinskiy, @acherkashin, @AndreyFedyukin, @Hailmary-CK, @webdanka, @WishMaster2310
